src/plugin/manager.js

```diff
@@ -165,9 +176,33 @@ class Plugins {
 
         if (aDependencies.includes(b.name)) return 1;
         else if (bDependencies.includes(a.name)) return -1;
-        
+
         return 0;
     }
+
+    /**
+     * Install the dependencies of a given plugin
+     * @param {Plugin} plugin The plugin to prepare
+     */
+    async npmInstall(plugin) {
```

Comment:
The function is async even though it doesn't have to be, although it would be nice for it to be async (so it won't block stuff), by using the async versions of writeFile and exec. And we could also maybe add some error checking to verify if the npm dependencies install correctly and if not, not start the whole dependency.

Reply:
Ooh, interesting. I'll make the changes now

Reply:
It seems that writeFile and exec are callback based, not promise based. If we keep it async, so that nothing is blocked, then we should still be ok? We could also convert the functions to promise based? What do you think?

Reply:
**Update**

I've promisified the writeFile and exec functions :smile:. However, when implementing a system to not load plugins with missing dependencies, but still load others, I ran into a few issues with the current `getTrueDependencies` implementation. Still, I should be able to finish this today once I decide what the cleanest way to implement it is.

Reply:
Added the code. I've also moved some logging as we were logging events that were not occuring